generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/ReductiveState.java

```diff
@@ -4,108 +4,103 @@
 import com.scottlogic.deg.generator.ProfileFields;
 
 import java.util.*;
-import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class ReductiveState {
 
     private final ProfileFields fields;
     private final Map<Field, FixedField> fixedFields;
-    private final FixedField lastFixedField;
+    private final FixedField nextFieldToFix;
```

Comment:
This name makes me feel uneasy. Whilst `lastFixedField` may be equally not-great, `nextFieldToFix` has the connotations of _the-field-that-will-be-fixed-next_, not _the-field-that-has-been-selected-but-value-not-pinned-yet_.

Not entirely sure on the right name here, but maybe one of the following might help to remove confusion (in future):

- `lastSelectedField`
- `lastField`

_last_ makes sense as it is something that **has** happened (it has been chosen/selected/~'fixed'~), albeit not necessarily pinned to a value yet.

My additional concern is that there will be periods of time where this field (whatever it is named) has a value fixed for it.

My preference would be for `lastSelectedField` (or something equivalent) as it does not imply the field is fix (and equally importantly that it is not fixed).